backend/mechanismbackend/api/views.py

```diff
@@ -51,3 +56,69 @@ def create(self, request, *args, **kwargs):
             content = {'error': str(e)}
             return Response(content, status=status.HTTP_400_BAD_REQUEST)
 
+
+# dirty hack, try to find better way to achieve this
+class MechanismMatrix(APIView):
+    def get(self, request, format=None):
+        predicate = self.get_query_predicate()
+
+        matrix = []
+        matrix_inner = []
+
+        with connection.cursor() as cursor:
+            for input in ['R', 'T']:
+                for i in range(1, 4):
+                    i = str(i)
+                    for output in ['R', 'T']:
+                            for o in range(1, 4):
+                                o = str(o)
+
+                                cursor.execute('select input' + input + i + ', output' + output + o + ', Count(*) from api_mechanism ' +
+                                               predicate + ' group by input' + input + i + ', output' + output + o)
+                                rows = cursor.fetchall()
+
+                                num_mechanisms = 0
+                                for tpl in rows:
+                                    tpl_in, tpl_out, count = tpl
+                                    if tpl_in and tpl_out:
+                                        num_mechanisms = count
+                                        break
+
+                                matrix_inner.append(num_mechanisms)
+                                if output + o == 'T3':
+                                    total_for_input = sum(matrix_inner)
+                                    matrix_inner.append(total_for_input)
+                                    matrix.append(matrix_inner)
+                                    matrix_inner = []
+
+
+            total_for_outputs = []
+            for output in range(6):
+                total_inner = 0
+                for input in range(6):
+                    total_inner += matrix[input][output]
+                total_for_outputs.append(total_inner)
+            total_for_outputs.append(0)  # pad to match dimensions
```

Comment:
Could you please fill this cell with the total number of mechanisms? This information might be helpful for the user.